spec/helpers/tasklist_helper_spec.rb

```diff
@@ -148,37 +143,26 @@
 
     context "with lettings log" do
       context "when collection_period_open? == true" do
-        context "with 2023 deadline" do
-          let(:now) { Time.utc(2022, 6, 1) }
-          let(:lettings_log) { create(:lettings_log, :completed) }
+        let(:now) { Time.utc(2022, 6, 1) }
+        let(:lettings_log) { build(:lettings_log, :completed, startdate: now, id: 123) }
 
-          it "returns relevant text" do
-            expect(review_log_text(lettings_log)).to eq(
-              "You can #{govuk_link_to 'review and make changes to this log', review_lettings_log_path(lettings_log)} until 9 June 2023.".html_safe,
-            )
-          end
+        before do
+          allow(lettings_log.form).to receive(:submission_deadline).and_return(Time.utc(2023, 6, 9))
         end
 
-        context "with 2024 deadline" do
-          let(:now) { Time.utc(2023, 6, 20) }
-          let(:lettings_log) { create(:lettings_log, :completed, national: 18, waityear: 2) }
-
-          it "returns relevant text" do
-            expect(review_log_text(lettings_log)).to eq(
-              "You can #{govuk_link_to 'review and make changes to this log', review_lettings_log_path(lettings_log)} until 7 June 2024.".html_safe,
-            )
-          end
+        it "returns relevant text" do
+          expect(review_log_text(lettings_log)).to eq(
+            "You can #{govuk_link_to 'review and make changes to this log', review_lettings_log_path(lettings_log)} until 9 June 2023.".html_safe,
+          )
         end
       end
 
       context "when collection_period_open? == false" do
-        let(:now) { Time.utc(2022, 6, 1) }
-        let!(:sales_log) { create(:lettings_log, :completed) }
+        let(:now) { Time.utc(2024, 6, 1) }
```

Comment:
as a general question for the whole PR and what we're trying to do with the tests overall, I was under the impression we would try and remove things where the time is fixed in general.
for example here it looks to me that we're testing that some text is different when a collection period is closed, it seems the most robust to me to stub the call to collection_period_open? is it possible to stub methods on a model made from a factory or only an instance double?

Reply:
Yep, can do that here, nice spot! Any specific tests in this PR that you noticed a similar issue on?

Reply:
no more a general comment. I can't see how making Time return a given date for now makes the tests more robust. Won't we still get the same type of problems when we roll over collection years?

Reply:
Yeah, that's fair. I think this might have been fine cause we're not reinstantiating the forms, but I've mocked some things now and removed the time travelling! For some suites the tests are still time dependent, so I've changed to stubbing `now` instead of freezing, cause we don't have to unfreeze it that way 🤷